src/neuroconv/tools/nwb_helpers/_configuration_models/_base_dataset_io.py

```diff
@@ -93,7 +93,7 @@ class DatasetIOConfiguration(BaseModel, ABC):
     model_config = ConfigDict(validate_assignment=True)  # Re-validate model on mutation
 
     # Immutable fields about the dataset
-    object_id: str = Field(description="The UUID of the neurodata object containing the dataset.", frozen=True)
+    object_id: str = Field(description="The UUID of the neurodata object containing the dataset.")
```

Comment:
Is this still needed anymore? @h-mayorquin

Reply:
This is needed:

https://github.com/catalystneuro/neuroconv/blob/7744c669cbb49078da40ae89650b93f2910bcdb1/src/neuroconv/tools/nwb_helpers/_configuration_models/_base_backend.py#L132-L135

Now that I look at it this looks as it is modifying the old configuration.

Let me fix this. Do you mind if I push here or would you rather have me make another PR?

Reply:
Did it